tools/ldb_cmd.cc

```diff
@@ -2166,7 +2189,7 @@ void DBDumperCommand::DoCommand() {
         // TODO(myabandeh): allow configuring is_write_commited
         DumpWalFile(options_, path_, /* print_header_ */ true,
                     /* print_values_ */ true, true /* is_write_commited */,
-                    &exec_state_);
+                    ucmps_, &exec_state_);
```

Comment:
One general question here: we now allow changing the comparator e.g. to enable UDTS on an existing CF, so the current comparator might be different from what was in effect when a given WAL record was written, right? If so, would it make sense to have some sanity checks around timestamp sizes (using the TS size info from the WAL) before using the current comparator? (This only affects `dump_wal` I think; for `dump` and `idump`, recovered KVs would go through the timestamp reconciliation logic.)

Reply:
This is a good point! I have introduced using the `HandleWriteBatchTimestampSizeDifference` util function with `kVerifyConsistency` mode when DB is opened and running comparators are provided. To check any user key format discrepancy between the comparator's expectation and WAL file's recorded timestamp size.

The current handling is we don't proceed if any discrepancy is detected and message like this will be printed:
```Failed: Format for user keys in WAL file is inconsistent with the comparator used to open the DB. Timestamp size recorded in WAL vs specified by comparator: {(cf_id: 0, [recorded: 8, comparator: 0])}```

I found that this PR's change introduced usage of `ColumnFamilyCollector` and `TimestampRecoveryHandler` for ldb's wal dump path. And those two were missing override implementations for both `PutEntityCF` and `TimedPutCF`. So I added those in this PR too.

Reply:
To quote Ron Burgundy, "that escalated quickly"! Thanks so much for adding these